client/app/dbaas/logs/detail/streams/add/logs-streams-add.controller.js

```diff
@@ -1,7 +1,84 @@
 class LogsStreamsAddCtrl {
-    constructor ($stateParams) {
+    constructor ($q, $state, $stateParams, LogsStreamsService, ControllerHelper, CloudMessage) {
+        this.$q = $q;
+        this.$state = $state;
         this.$stateParams = $stateParams;
         this.serviceName = this.$stateParams.serviceName;
+        this.LogsStreamsService = LogsStreamsService;
+        this.ControllerHelper = ControllerHelper;
+        this.CloudMessage = CloudMessage;
+        this.isEdit = false;
+        this.compressionAlgorithms = this.LogsStreamsService.getCompressionAlgorithms();
+        this.storageDurations = this.LogsStreamsService.getStorageDurations();
+
+        this.initLoaders();
+    }
+
+    /**
+     * initializes options list
+     *
+     * @memberof LogsStreamsHomeCtrl
+     */
+    initLoaders () {
+        this.options = this.ControllerHelper.request.getArrayLoader({
+            loaderFunction: () => this.LogsStreamsService.getSubscribedOptions(this.serviceName)
+        });
+        this.options.load();
+
+        if (this.$stateParams.streamId) {
+            this.isEdit = true;
+            this.stream = this.ControllerHelper.request.getHashLoader({
+                loaderFunction: () => this.LogsStreamsService.getStream(this.serviceName, this.$stateParams.streamId)
+            });
+            this.stream.load();
+        } else {
+            this.isEdit = false;
+            this.stream = {
+                data: {
```

Comment:
I like tu put default initialization in my gets / service methods but I know not eveyrone does

Reply:
its good practice. moved it to service